pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/0defa141-bf03-43d3-8060-146cfb8af383?testReportRun=30&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [814a375d4c0e79d875c42452725f05f6c27294e3](https://github.com/cockroachdb/cockroach/commits/814a375d4c0e79d875c42452725f05f6c27294e3):


```
   {
    "query": "SELECT EXISTS (SELECT schema_name FROM information_schema.schemata WHERE schema_name = $1)",
    "queryArgs": [
     "schema_w0_21"
    ],
    "result": true
   }
  ],
  "previousStatements": [
   "DROP SCHEMA \"schema_w0_21\" CASCADE"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: unimplemented: cannot drop type "schemachange.schema_w0_21._enum_w1_56" because other objects ([schemachange.public.udf_w1_74]) still depend on it (SQLSTATE 0A000)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3113
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:462
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:544
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: unimplemented: cannot drop type "schemachange.schema_w0_21._enum_w1_56" because other objects ([schemachange.public.udf_w1_74]) still depend on it (SQLSTATE 0A000)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 157
Total Schema Statements Succeeded = 157
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 88
Total Transactions Rolled Back = 77
Total Transactions Executed = 165
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_30_of_30/test.outputs/logTestWorkload827641397
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload827641397
--- FAIL: TestWorkload (23.76s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=30</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36041

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13976553?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13976553?buildTab=artifacts#/) on master @ [814a375d4c0e79d875c42452725f05f6c27294e3](https://github.com/cockroachdb/cockroach/commits/814a375d4c0e79d875c42452725f05f6c27294e3):


```
     ]
    ]
   },
   {
    "query": "SELECT ARRAY[parent.table_schema, parent.table_name, parent.column_name, child.column_name] FROM (SELECT conname, conkey, confkey, conrelid, confrelid FROM pg_constraint WHERE (contype = 'f') AND (conrelid = 'schema_w1_84.table_w1_98'::REGCLASS::INT8)) AS con JOIN (SELECT constraint_name FROM information_schema.table_constraints WHERE ((table_schema = 'schema_w1_84') AND (table_name = 'table_w1_98')) AND (crdb_internal.is_constraint_active('schema_w1_84.table_w1_98', constraint_name) = true)) AS tc ON conname = tc.constraint_name JOIN (SELECT column_name, ordinal_position, column_default FROM information_schema.columns WHERE (table_schema = 'schema_w1_84') AND (table_name = 'table_w1_98')) AS child ON conkey[1] = child.ordinal_position JOIN (SELECT pc.oid, cols.table_schema, cols.table_name, cols.column_name, cols.ordinal_position FROM pg_class AS pc JOIN pg_namespace AS pn ON pc.relnamespace = pn.oid JOIN information_schema.columns AS cols ON ((pc.relname = cols.table_name) AND (pn.nspname = cols.table_schema))) AS parent ON ((con.confkey[1] = parent.ordinal_position) AND (con.confrelid = parent.oid)) WHERE child.column_name != 'rowid'",
    "result": null
   }
  ],
  "previousStatements": [
   "INSERT INTO schema_w1_84.table_w1_98 (col98_w1_99,col98_w1_100,\" col9😍8_w1_101\",col98_w1_102,\"c%pol98_w1_103\",col98_w1_104,\"col%q98_w1_105\",col98_w1_106,col98_w1_107,\"%qcol98_w1_108\",\"co\\\\x1dl98_w1_109\",\"col%p9\\\\uAE128_w1_110\",col98_w1_111,col98_w1_112,😫col98_w1_113,\"cOl98_w1_114\") VALUES (NULL,(-0.013468781486153603):::FLOAT8,e'\\x14'::VARCHAR,'201444d8-adea-4394-9ca3-d7e4511f69db':::UUID,'05:23:23.548623+09:37':::TIMETZ,'01040000E0E61000000800000001010000C0D8BA4FE5971C5040D0FED4D4B73D27C0D49284D43477F4C14F0A1CBB9325F4C101010000C0E05A7844D3A25AC092577931DFBD3FC008F88E5F2355E2C13680D3F97E67F9C101010000C0D87C836A8C4C49C0204B6809A92407C008DB6665E3B6FD418087F2EF6AB4B8C101010000C0DA3F86C5DF1466C0D8C216615CCF4440F27FEF8CFE27F0416871F404ECE4EA4101010000C05018E5069E014E4039258171F23856C06C3AFD6D108CEBC16083CD259A99DCC101010000C09AAFA57F35AD5240C28C6B65064B52C012258F05783FF0C1786C93E0489AFF4101010000C038E06BFD33EE6440169269E8DB523FC0E29E86B6B60B0242204AC24F5197DAC101010000C08C314F5CB9835D40F8135886B8682A400A1455EEA292FBC16A6C7B8F31BAE4C1':::GEOGRAPHY,'''Do'' ''E'':59C,96C,170B ''LcuBvgzdff'' ''NmT'' ''Py'' ''Y'':124A,486B ''jchIL'':38A,257C,537A ''jxRxhTUu'' ''vY'':983B ''zo''':::TSVECTOR,'2007-06-02 17:32:01.000117+00':::TIMESTAMPTZ,(-1):::INT8,e'\\U0007D093\\U000E8FC3\\U000E7D65\\U000E4261' COLLATE da_DK,'1978-06-05':::DATE,8400156422939723262:::INT8,'''BpiI''':::TSQUERY,'''KmDZRoBG'' | ''CFhyyGnr'' \u003c-\u003e ''FtbmupX'' \u0026 ''hCRvqIo''':::TSQUERY,'\\xb712':::BYTES,1082:::OID),(NULL,(-0.032362643629312515):::FLOAT8,e'\\x1f;a\\x19B'::VARCHAR,'ee8d33a3-cee7-48a5-a1f5-c8ebed082f03':::UUID,'19:28:03.486867-04:46':::TIMETZ,'01030000E0E6100000010000000C00000031FD800B87F765C0791CFF71F24450C09ED0D43CE35FE2C1D8E8ABE80147EC41409D1E79167611C036E65F9CC3FF3CC070AC596589A9C2C1A6EDCE965F8702C20061FF15773E4340A7708F69E3B852C056ED4536A6E2F7C1F499DD87BF2B0042BC8DD0142C3A564006072E1A8E784DC03A2CC4A3974CF2410A6B503A494DF94110721AE98B3A5540DC08E88DD11F40C07002CE5AEE3CF2C13269A2D0DB47004238A98554EC5A654039EC489DDD8746C0E89E4193D270E54125DC6E0A2392F5C1BA8D939B869B63402825254FAB7E23C07C0C368F2313004238A1383B3C84D0C1702D2D88C09C23C080EE32D22C2D4840583202AEB278EE410A2E67C7E5BB014284074440301D60C044BC7AC293344B40905F6552DE0AF4C17E1D08CACE58F8C1B06F3DFBE2F664C08CC37787DDF14A404061D4BE4ADBA7C1B82CE357B6F0EA41F52659DFF13C61C040DF940ACF2E054024A3DFB070720042F343C7BB16A9F8C131FD800B87F765C0791CFF71F24450C09ED0D43CE35FE2C1D8E8ABE80147EC41':::GEOGRAPHY,'''CAp'' ''CtdqrRM'':272C,703B ''EkXWiIinyo'' ''JrRFpvWEjm'' ''aiyAfyfglk'':153C,515B,779B ''kmXyE'':618C,975A ''lb'' ''myMLv'':483A ''ppOtTkcQ'' ''q'':8B,437A,668B':::TSVECTOR,'1991-08-19 03:00:04.000983+00':::TIMESTAMPTZ,2047039450:::INT8,e'\\U00026186\\U0009D96F\\U00059416\\U0000A6EF\\U0001EF8E\\U000E0BF9\\U000AE238\\U000C7150\\U00063E86' COLLATE da_DK,'1973-09-23':::DATE,(-1787389379148348367):::INT8,'''vmWXSwMhqb'' | ''DZ'' \u003c-\u003e !''hUswtLR'' \u0026 !''rvSVeUYBt''':::TSQUERY,'!''djRtWjnIgU''':::TSQUERY,'\\x52a302ed':::BYTES,700:::OID)"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: integer out of range (SQLSTATE 22003)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3113
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:462
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:544
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: integer out of range (SQLSTATE 22003)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 170
Total Schema Statements Succeeded = 170
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 90
Total Transactions Rolled Back = 102
Total Transactions Executed = 192
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2198885889
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2198885889
--- FAIL: TestWorkload (133.95s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13992008?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13992008?buildTab=artifacts#/) on master @ [254dbd247fb8ed352a11439063b29f23a0767f28](https://github.com/cockroachdb/cockroach/commits/254dbd247fb8ed352a11439063b29f23a0767f28):


```
     ]
    ]
   },
   {
    "query": "SELECT ARRAY[parent.table_schema, parent.table_name, parent.column_name, child.column_name] FROM (SELECT conname, conkey, confkey, conrelid, confrelid FROM pg_constraint WHERE (contype = 'f') AND (conrelid = 'public.table_w1_35'::REGCLASS::INT8)) AS con JOIN (SELECT constraint_name FROM information_schema.table_constraints WHERE ((table_schema = 'public') AND (table_name = 'table_w1_35')) AND (crdb_internal.is_constraint_active('public.table_w1_35', constraint_name) = true)) AS tc ON conname = tc.constraint_name JOIN (SELECT column_name, ordinal_position, column_default FROM information_schema.columns WHERE (table_schema = 'public') AND (table_name = 'table_w1_35')) AS child ON conkey[1] = child.ordinal_position JOIN (SELECT pc.oid, cols.table_schema, cols.table_name, cols.column_name, cols.ordinal_position FROM pg_class AS pc JOIN pg_namespace AS pn ON pc.relnamespace = pn.oid JOIN information_schema.columns AS cols ON ((pc.relname = cols.table_name) AND (pn.nspname = cols.table_schema))) AS parent ON ((con.confkey[1] = parent.ordinal_position) AND (con.confrelid = parent.oid)) WHERE child.column_name != 'rowid'",
    "result": null
   }
  ],
  "previousStatements": [
   "INSERT INTO public.table_w1_35 (col35_w1_36,\"\u000bcol35_w1_37\",col35_w1_38,😤col3😭5_w1_39,\"c ol35_w1_40\",col35_w1_41,\"cOl35_w1_42\",col35_w1_43,col35_w1_44,col35́_w1_45,col35_w1_46,\"c%qol3!5_w1_47\",\"col3 %q5_w1_48\") VALUES ('''cjXOk''':::TSQUERY,'j'::CHAR,'63 years 7 mons 616 days 16:14:20.520148':::INTERVAL,'2031-02-22 17:14:54.000866+00':::TIMESTAMPTZ,'01040000C00700000001010000C0103DC81CA258D7C1DC3DC430F11CE3C174657E27B921DBC1AE16B3176314F6C101010000C00DAEDDB0291CF3C140D7240F815801425A9A02051AFEF2C1FC0EB96F6F71DBC101010000C0704A53309B30FFC1C062E00724B4FB41D05929BF30F0D34174F7B8357C7CE74101010000C040CF8FDC3C88FAC14834EFD7037AD8C13A3E211B883802C268CDCCB40D9BF74101010000C0601993D9173BED41B884C6CA1094F24104E8AFAE31A0DDC1E51ED3916555F0C101010000C0CACA23C08FB0FBC1C66830358893F34118F6DC411FF8E54128C9EB331F17D94101010000C0D82A93B8917CEBC1199B7599D74201C26867A20E06FCD941F8E89CB65B6DDF41':::GEOMETRY,(-6320100189472250654):::INT8,'14:30:49.644847':::TIME,4096:::OID,'%_,G``|'::NAME:::NAME,(-21653):::INT8,'BOX(-0.126727328187925 -0.8302947143767121,0.6542174872457518 -0.08397357541037781)':::BOX2D,'[true, {\"bar\": 0.6674944800553222, \"qS\": 0.48892052476584813}, true, false]':::JSONB,'|'::CHAR)"
  ]
 }
}
    schema_change_external_test.go:132: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:132
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: integer out of range (SQLSTATE 22003)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:3113
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:463
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:545
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schemachange_test_test/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:121
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: integer out of range (SQLSTATE 22003)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 71
Total Schema Statements Succeeded = 71
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 39
Total Transactions Rolled Back = 54
Total Transactions Executed = 93
    schema_change_external_test.go:98: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2094256356
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/fcba020ca7cb44bf8385bd0e13bcbe2d/logTestWorkload2094256356
--- FAIL: TestWorkload (60.24s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118879 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.1.15-rc]
- #117782 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-3 T-sql-foundations branch-release-23.2.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -2860,6 +2860,15 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 			if col.typ.Family() == types.Oid.Family() {
 				d = tree.NewDOid(randgen.RandColumnType(og.params.rng).Oid())
 			}
+			// We have seen cases where randomly generated ints easily hit an
+			// integer overflow in our workload when we allow large numbers.
+			// Since there is no real advantage to testing such numbers,
+			// limit the largeness by always setting the amount of bits to
+			// 8 (-128 to 127) - ensuring we won't overflow even with the
+			// smallest int (INT2, -32768 to 32767).
+			if col.typ.Family() == types.IntFamily {
+				d = tree.NewDInt(tree.DInt(int8(og.params.rng.Uint64())))
+			}
 			str := tree.AsStringWithFlags(d, tree.FmtParsable)
 			// For strings use the actual type, so that comparisons for NULL values are sane.
 			if col.typ.Family() == types.StringFamily {

